src/events/tls.py

```diff
@@ -135,8 +136,12 @@ def _trusted_relation_created(self, event: EventBase) -> None:
             event.defer()
             return
 
-        # Create a "mtls" flag so a new listener (CLIENT_SSL) is created
-        self.charm.state.cluster.update({"mtls": "enabled"})
+        if not self.charm.state.cluster.mtls_enabled:
+            # Create a "mtls" flag so a new listener (CLIENT_SSL) is created
+            self.charm.state.cluster.update({"mtls": "enabled"})
+            self.charm.on.config_changed.emit()
+
+        self.charm.app.status = ActiveStatus()
```

Comment:
**todo**: Similar to my new comment on VM, could you check if we really need this statement?

Reply:
Yep, it's redundant and could be removed.